compiler/rustc_builtin_macros/src/asm.rs

```diff
@@ -524,8 +522,16 @@ fn expand_preparsed_asm(
             ast::StrStyle::Raw(raw) => Some(raw as usize),
         };
 
-        let template_str = &template_str.as_str();
         let template_snippet = ecx.source_map().span_to_snippet(template_sp).ok();
+        template_strs.push((
+            template_str,
+            // This should never fail, every template that exists has a span and has a souce snippet that is part of that span
```

Comment:
I think this can fail if the span comes from another crate for which we only have a rlib and no source code. This would be something like a `macro_rules!` in that crate which calls `asm!` with a string literal, and that macro is invoked from the current crate.

Reply:
ah :/